### PR TITLE
Header search refactoring

### PR DESCRIPTION
**Pull request by Hayk-web**
Changed custom forms logic to "Drupal"  way for the header search form and integrate with existing facets.

### CODE REVIEW

File: config/default/views.view.liiweb_search.yml [hunk 1]
```diff
@@ -530,9 +574,10 @@ display:
             group_items: {  }
           plugin_id: search_api_date
       filter_groups:
-        operator: AND
+        operator: OR
```
Comment (longhotsummer) [inline config/default/views.view.liiweb_search.yml:577]: @mariyabb I just want to confirm this (I don't fully understand it): does this change the default operator to OR from AND @Hayk-web ?

Comment (longhotsummer) [reply config/default/views.view.liiweb_search.yml:577]: Or the line just below...

**Review by arthurbaghdas — APPROVED** [review]

File: config/default/views.view.liiweb_search.yml (hunk 1 above)
Comment (Hayk-web) [reply config/default/views.view.liiweb_search.yml:577]: The logic is the same. Full text and content type (search on the header)  criteria work together by AND operator and on the search page, the date search works with them with OR operator. 
image

Comment (mariyabb) [reply config/default/views.view.liiweb_search.yml:577]: @Hayk-web The only anomaly I see is on the search result page, if I add a Custom Date range and hit Apply, the search filters do update but the search results do not automatically update. I have to click on the Date filter to get the search results to display the refined query. 

But the search logic on the full-text search is intact.

Comment (Hayk-web) [reply config/default/views.view.liiweb_search.yml:577]: I have checked and this issue isn't related to changes that have been done and it actually on the live site too. The problem is the date format you need to fill like so 2017-03-16 12:00:00. Would you want to fix this in the same task, or will open the separate one?

**Pull request merged**